### ISSUE TITLE
GH Actions: run seems to run with noglob

### ISSUE THREAD

**Issue by jf** [actions/virtual-environments#4121]
### Description

We have the following in a "run" step in github actions:

```
BUCKET=<bucket name>
COMMIT=commit_*
aws s3 cp sgfe.tbz s3://$BUCKET/latest.tbz
aws s3 cp sgfe.tbz s3://$BUCKET/$COMMIT.tbz
```

The output for github actions (`shell: /usr/bin/bash -e {0}`) seems to imply that bash is run, while the documentation at https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsrun indicates `bash --noprofile --norc -eo pipefail {0}`. Both of them do NOT have noglob activated, and yet we find that $COMMIT is set to the literal `commit_*`, instead of the actual (for example) commit_b6ff060a13995a177f5ef36508a79f620675f844 that is present on disk.

### Virtual environments affected

- [ ] Ubuntu 16.04
- [ ] Ubuntu 18.04
- [X] Ubuntu 20.04
- [ ] macOS 10.15
- [ ] macOS 11
- [ ] Windows Server 2016
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

20210906.1

### Repro steps

Create a "commit_{any suffix}" file, and then in a run step, use the following lines:

```
COMMIT=commit_*
echo $COMMIT
```

Instead of the actual "commit_" filename, you get a literal `commit_*`

**Comment by al-cheb** [comment]
Hello, @jf
Are you sure that the file exists?

```
- run: |
        touch commit_test
        COMMIT=commit_*
        echo $COMMIT
```
        
![image](https://user-images.githubusercontent.com/47745270/134119169-9c3e40cb-b4b6-4297-b6d9-f0576483bb58.png)

**Comment by jf** [comment]
Yes. It does come from an earlier `run` step, though. But I've assumed that it does exist. The earlier `run` creates 2 files: `sgfe.tbz`, and the "commit_*" file. I've assumed that there is no issue with the "commit_*" file because the sgfe.tbz file is also there when we do the `aws s3 cp`

**Comment by al-cheb** [comment]
@jf, Globbing happens after variable expansion. You should use: `COMMIT=$(echo commit_*)` to store the result of expansion.

```
    steps:
    - run: |
        touch fixed_filename
        touch commit_test
    - run: |
         echo "Default set"
         COMMIT=commit_*
         echo $COMMIT
         echo "Enable filename expansion (globbing)."
         set +f 
         COMMIT=commit_*
         echo $COMMIT
         echo "Disable filename expansion (globbing)."
         set -f 
         COMMIT=commit_*
         echo $COMMIT
```

![image](https://user-images.githubusercontent.com/47745270/134124224-1dae3e9f-3345-48e5-abdc-5c40bd0140a6.png)


Your example:
![image](https://user-images.githubusercontent.com/47745270/134124475-950a8c5b-03f9-4542-bde1-6707aa7c6488.png)

**Comment by jf** [comment]
Thank you, @al-cheb . I'll try the $() fix. I have to say though that this is still a mystery to me. We've never added, or changed the 'set -/+f' setting, and it's as per default, so I still don't understand why it would not work.

**Comment by al-cheb** [comment]
> Thank you, @al-cheb . I'll try the $() fix. I have to say though that this is still a mystery to me. We've never added, or changed the 'set -/+f' setting, and it's as per default, so I still don't understand why it would not work.

I see the issue with globbing is still not resolved - https://github.com/aws/aws-cli/issues/925

**Comment by jf** [comment]
> > Thank you, @al-cheb . I'll try the $() fix. I have to say though that this is still a mystery to me. We've never added, or changed the 'set -/+f' setting, and it's as per default, so I still don't understand why it would not work.
> 
> I see the issue with globbing is still not resolved - [aws/aws-cli#925](https://github.com/aws/aws-cli/issues/925)

thanks for the helpful note! apologies, but I closed this without seeing this reply. In this case we can work around the absence of globbing with aws-cli, but this is good to know. Thank you once again!
